core/integration/bookkeeping/plugin.go

```diff
@@ -117,30 +118,60 @@ func (p *Plugin) CallStack(data interface{}) (stack map[string]interface{}) {
 	varStack := call.VarStack
 	envId, ok := varStack["environment_id"]
 	if !ok {
-		log.Error("cannot acquire environment ID")
+		err := errors.New("cannot acquire environment ID")
+		log.Error(err)
+
+		call.VarStack["__call_error_reason"] = err.Error()
+		call.VarStack["__call_error"] = "Bookkeeping plugin Call Stack failed"
+		return
+	}
+	trigger, ok := varStack["__call_trigger"]
+	if !ok {
+		err := errors.New("cannot acquire trigger from varStack")
+		log.WithField("call", call).
+			WithField("partition", envId).
+			Error(err)
+
+		call.VarStack["__call_error_reason"] = err.Error()
+		call.VarStack["__call_error"] = "Bookkeeping plugin Call Stack failed"
 		return
 	}
 	var err error
 	parsedEnvId, err := uid.FromString(envId)
 	if err != nil {
+		if strings.Contains(trigger, "DESTROY") || strings.Contains(trigger, "GO_ERROR") {
```

Comment:
Shouldn't these (multiple occurences) also populate the `__call_error` and `__call_error_reason`?

Reply:
No, these should not, it's one of the points of the PR